### PR TITLE
Unit Test for Update

### PR DESCRIPTION
**Pull request by Treesa-James**
Unit test/update for all standard objects

### CODE REVIEW

File: tests/IntegrationTest.php [hunk 1]
```diff
@@ -479,18 +479,149 @@ function testExtract()
...
+        $schema = json_decode(file_get_contents(__DIR__ . "/schemas/DiscoverResult.json"), true);
+        $integration->setSchema(new IntegrationSchema($schema));
+        $integration->begin();
+
+        // Define the query condition for extracting records
+        $query = ["where" => ['left' => 'hs_object_id', 'op' => '=', 'right' => '21585771784']];
+        $recordLocator = new RecordLocator(["recordType" => 'companies', "query" => $query, 'type' => OperationTypes::Update]);
+        // Define the mapping for the update
+        $mapping = new Mapping([
+            'name'   => 'Test',
```
Comment (arun-experion) [inline tests/IntegrationTest.php:512]: Use a valid name

File: tests/IntegrationTest.php [hunk 2]
```diff
@@ -479,18 +479,149 @@ function testExtract()
...
+        $schema = json_decode(file_get_contents(__DIR__ . "/schemas/DiscoverResult.json"), true);
+        $integration->setSchema(new IntegrationSchema($schema));
+        $integration->begin();
+
+        // Define the query condition for extracting records
+        $query = ["where" => ['left' => 'hs_object_id', 'op' => '=', 'right' => '33013881138']];
+        $recordLocator = new RecordLocator(["recordType" => 'contacts', "query" => $query, 'type' => OperationTypes::Update]);
+        // Define the mapping for the update
+        $mapping = new Mapping([
+            'firstname'   => 'Test",
```
Comment (arun-experion) [inline tests/IntegrationTest.php:545]: Use a valid name

File: tests/IntegrationTest.php [hunk 3]
```diff
@@ -479,18 +479,149 @@ function testExtract()
...
+        $integration->setSchema(new IntegrationSchema($schema));
+        $integration->begin();
+
+        // Define the query condition for extracting records
+        $query = ["where" => ['left' => 'hs_object_id', 'op' => '=', 'right' => '20651824971']];
+        $recordLocator = new RecordLocator(["recordType" => 'deals', "query" => $query, 'type' => OperationTypes::Update]);
+        // Define the mapping for the update
+        $mapping = new Mapping([
+            'amount'   => '100',
+            "dealname" => "Test deal",
```
Comment (arun-experion) [inline tests/IntegrationTest.php:580]: Use a valid name

File: tests/IntegrationTest.php [hunk 4]
```diff
@@ -479,18 +479,149 @@ function testExtract()
...
+        $integration->begin();
+
+        // Define the query condition for extracting records
+        $query = ["where" => ['left' => 'hs_object_id', 'op' => '=', 'right' => '2946238838']];
+        $recordLocator = new RecordLocator(["recordType" => 'tickets', "query" => $query, 'type' => OperationTypes::Update]);
+        // Define the mapping for the update
+        $mapping = new Mapping([
+            "hs_pipeline_stage" => "1",
+            "hs_ticket_priority" => "HIGH",
+            'subject'   => 'Test Ticket'
```
Comment (arun-experion) [inline tests/IntegrationTest.php:614]: Use a valid name

File: tests/IntegrationTest.php [hunk 5]
```diff
@@ -84,7 +84,7 @@ function testDiscoverReturnsJsonSchema()
 
         // Verify that the 'items' array contains at least 4 elements
         //To Ensure that all standard objects are present in schema
-        $this->assertGreaterThanOrEqual(4, count($schema->schema['items']), 'The items array should contain at least 4 arrays.');
+        $this->assertGreaterThan(4, count($schema->schema['items']), 'The items array should contain at least 4 arrays.');
```
[the remark this answers is not in the document]
Comment (Treesa-James) [reply tests/IntegrationTest.php]: corrected

File: tests/IntegrationTest.php [hunk 6]
```diff
@@ -364,7 +364,7 @@ function testLoadDeals()
         //Check the missing required fields
         $this->assertTrue($mapping->hasItem('dealstage'));
         $this->assertTrue($mapping->hasItem('dealname'));
-        
+
```
[the remark this answers is not in the document]
Comment (Treesa-James) [reply tests/IntegrationTest.php]: corrected

File: tests/IntegrationTest.php (hunk 3 above)
Comment (Treesa-James) [reply tests/IntegrationTest.php]: corrected

File: tests/IntegrationTest.php (hunk 4 above)
Comment (Treesa-James) [reply tests/IntegrationTest.php]: corrected

**Review by arun-experion — APPROVED** [review]
code review completed

**Pull request merged**